### PR TITLE
Use best Rust practices

### PR DESCRIPTION
**Pull request by wmitros**
cargo-fmt and cargo-clippy are the useful tools for improving Rust code. We should provide them and recommend developers to use them.

This patch also includes a few small fixes recommended by these tools

### CODE REVIEW

File: install-dependencies.sh [hunk 1]
```diff
@@ -104,6 +104,8 @@ fedora_packages=(
     curl
     rust
     cargo
+    rustfmt
+    clippy
```
Comment (avikivity) [inline install-dependencies.sh:108]: Please split this patch into a separate pull request. Changes to install-dependencies.sh require extra handling and should come alone (the change to rust.md can remain in the patch)

**Review by piodul — COMMENTED** [review]
If we want to recommend the use of `cargo format` and `cargo clippy`, I think we should go all in and _enforce_ the use of those tools. If only a fraction of the developers adhere to our recommendation, then the remaining developers will give them a hard time by committing code that does not conform to the standards enforced by the tools.

We discussed this in our team's meeting. Based on the discussion, I propose to start with just verifying that the code is properly formatted as a part of the build process - so that both local build and CI should fail if the code is not properly formatted.

As for `cargo clippy`, we should be more careful with it. New versions introduce new lints that are enabled by default, and upgrading the version of the Rust toolchain can sometimes cause the code not to pass the clippy check anymore - that's a regular annoyance of ours in scylla-rust-driver, but we accept it and welcome the new lints as an opportunity to improve the style of the code. I'm not sure it would be a good idea for the scylladb repo, though. Maybe we could disable all default lints and opt-in to some of the current ones - not ideal because somebody has to remember to add more lints over time, but better than no lints at all.

@avikivity what do you think?

**Comment by piodul** [comment]
@wmitros Can we close this PR? We don't have plans to add more Rust code in the near future, so I don't think it makes sense to keep this open or spend resources on rebasing it in order to merge it.

**Comment by wmitros** [comment]
> @wmitros Can we close this PR? We don't have plans to add more Rust code in the near future, so I don't think it makes sense to keep this open or spend resources on rebasing it in order to merge it.

Makes sense. There's not much decided here and if we ever continue developing Rust we'll need to consider the approach for it from the ground up anyway so I'll close this PR
